core/src/components/input/input.tsx

```diff
@@ -771,6 +792,30 @@ export class Input implements ComponentInterface {
                 <ion-icon aria-hidden="true" icon={mode === 'ios' ? closeCircle : closeSharp}></ion-icon>
               </button>
             )}
+            {this.showPasswordToggle && !readonly && !disabled && (
+              <button
+                aria-label="show password"
```

Comment:
This aria label should change based on whether or not the password is visible

Reply:
I think this is not the case when we use the `switch` role. If the aria-label changes, the screen reader will read: 
> show password on
> hide password off

I don't know if it would be completely wrong, but it isn't very clear to me.

If we want the aria-label to change, I should drop the `switch` role and use it as a simple button. Both approaches are commonly used, and I'm fine with either.

Reply:
Oh true that would be confusing 🤔 Ok let's keep it as-is for now.